zenodo/modules/jsonschemas/compilers.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Zenodo.
+# Copyright (C) 2016 CERN.
+#
+# Zenodo is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Zenodo is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""JSON schema compiler methods."""
+
+from __future__ import absolute_import, print_function
+
+from .utils import merge_dicts, remove_keys, resolve_schema_url, \
+    resolve_schema_path, replace_schema_refs
+
+
+def _iter_all_of(schema):
+    """Iterate over the items within schema 'allOf' definition at the root."""
+    if 'allOf' in schema:
+        for sub_schema_ref in schema['allOf']:
+            sub_schema_url = sub_schema_ref['$ref']
+            sub_schema = resolve_schema_url(sub_schema_url)
+            yield sub_schema
+
+
+def _compile_common(schema):
+    """Compile common parts for record and deposit."""
+    id_ = schema['id']
+    title = schema['title']
+    # We need to iter 'allOf' manually because jsonresolver
```

Comment:
Since I do not care about loaded-ordering, as I compile jsonschemas into a key-softed this is no longer true, but I'd not want to spend time fixing this right now as this solution works.